test/source/tests/settings.ts

```diff
@@ -427,6 +510,40 @@ export let defineSettingsTests = (testVariant: TestVariant, testWithBrowser: Tes
       await settingsPage1.close();
     }));
 
+    ava.default('settings - email change', testWithBrowser('ci.tests.gmail', async (t, browser) => {
+      const acct1 = 'ci.tests.gmail@flowcrypt.test';
+      const acct2 = 'user@forbid-storing-passphrase-org-rule.flowcrypt.test';
+      const settingsPage = await browser.newPage(t, TestUrls.extensionSettings(acct1));
+      const { cryptup_citestsgmailflowcrypttest_rules: oldRules, cryptup_citestsgmailflowcrypttest_passphrase_07481C8ACF9D49FE: savedPassphrase1 } =
+        await settingsPage.getFromLocalStorage(['cryptup_citestsgmailflowcrypttest_rules', 'cryptup_citestsgmailflowcrypttest_passphrase_07481C8ACF9D49FE']);
+      expect(savedPassphrase1).not.to.be.an('undefined');
+      expect((oldRules as { flags: string[] }).flags).not.to.include('FORBID_STORING_PASS_PHRASE');
+      await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+      const experimentalFrame = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, '@action-open-module-experimental', ['experimental.htm']);
+      await experimentalFrame.waitAndClick('@action-change-email');
+      const oauthPopup1 = await browser.newPageTriggeredBy(t, () => PageRecipe.waitForModalAndRespond(experimentalFrame, 'confirm',
+        { contentToCheck: 'email address has changed', clickOn: 'confirm' }));
+      await OauthPageRecipe.mock(t, oauthPopup1, acct2, 'override_acct');
+      await PageRecipe.waitForModalAndRespond(experimentalFrame, 'confirm',
+        { contentToCheck: 'email from ci.tests.gmail@flowcrypt.test to user@forbid-storing-passphrase-org-rule.flowcrypt.test', clickOn: 'confirm' });
+      const newSettingsPage = await browser.newPageTriggeredBy(t, () => PageRecipe.waitForModalAndRespond(experimentalFrame, 'info',
+        { contentToCheck: 'Email address changed to user@forbid-storing-passphrase-org-rule.flowcrypt.test', clickOn: 'confirm' }));
+      await Util.sleep(2);
+      // await PageRecipe.waitForModalAndRespond(?, 'confirm',
+      //   { contentToCheck: 'Your email aliases on Gmail have refreshed since the last time you used FlowCrypt', clickOn: 'confirm' });
+      const { cryptup_userforbidstoringpassphraseorgruleflowcrypttest_rules: newRules,
+        cryptup_userforbidstoringpassphraseorgruleflowcrypttest_keys: keys,
+        cryptup_userforbidstoringpassphraseorgruleflowcrypttest_passphrase_07481C8ACF9D49FE: savedPassphrase2 } =
+        await settingsPage.getFromLocalStorage(['cryptup_userforbidstoringpassphraseorgruleflowcrypttest_rules',
+          'cryptup_userforbidstoringpassphraseorgruleflowcrypttest_keys',
+          'cryptup_userforbidstoringpassphraseorgruleflowcrypttest_passphrase_07481C8ACF9D49FE']);
+      expect((newRules as { flags: string[] }).flags).to.include('FORBID_STORING_PASS_PHRASE');
+      expect((keys as KeyInfo[])[0].longid).to.equal('07481C8ACF9D49FE');
```

Comment:
so, if we forget 'keys' of older acct, the older `ci.tests.gmail@flowcrypt.test` key and passphrase get wiped out.
Is this the expected behaviour?

Reply:
That seems wrong - this way the new account has no keys. Old keys should still be transferred to the new account. I thought we employed some direct method to do it, and therefore it would make sense to use `forget`. Let me see.

Reply:
Shouldn't this take care of that?

```
    for (const ki of destAccountPrivateKeys) {
      await KeyStore.add(newAcctEmail, ki.private);
    }
    for (const fingerprint of Object.keys(destAcctPassPhrases)) {
      await PassphraseStore.set('local', newAcctEmail, fingerprint, destAcctPassPhrases[fingerprint]);
    }
```

Undefined in target `keys` is definitely wrong.

Reply:
Here is the bug:

```
const destAccountPrivateKeys = await KeyStore.get(newAcctEmail);
```

should use old account email here.

Reply:
Sorry, I see what you were trying to do - potentially preserving keys already potentially set up on target account.

I don't think you have to do that - since you store new keys one by one like this:

```
    for (const ki of destAccountPrivateKeys) {
      await KeyStore.add(newAcctEmail, ki.private);
    }
```

The keys should get merged. If one key gets overriden by another, then they both had the same fingerprint - not a problem. It could maybe be an older version that overwrites the newer one, but that's ok enough.

Reply:
Meaning, disregarding any keys already present on the target account and just saving the keys from source account in the target account blindly (with `await KeyStore.add(newAcctEmail, ki.private);`) should produce the desired result.